app/models/piece.rb

```diff
@@ -11,6 +11,10 @@ def user
     color == "white" ? game.white_player : game.black_player
   end
 
+  def is_move_malformed?(new_rank, new_file)
+    rank == new_rank && file == new_file
+  end
```

Comment:
I will move it down below the `protected` and let Ken know if he won't put this method under `protected'. thanks for the resource.